File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/CustomUIDelegate/CustomGroupDetailImageCell.m
//
//  CustomGroupDetailImageCell.m
//  DKImagePickerControllerDemo_OC
//
//  Created by zm on 2017/7/10.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "CustomGroupDetailImageCell.h"
#import "DKImageResource.h"
@interface CustomGroupDetailImageCell()
@property (nonatomic, strong) UIImageView * thumbnailImageView;
@property (nonatomic, strong) UIImageView * checkView;
@end


@implementation CustomGroupDetailImageCell
+ (NSString *)cellReuseIdentifier{
    return @"CustomGroupDetailImageCell";
}

- (UIImageView *)thumbnailImageView{
    if (!_thumbnailImageView) {
        _thumbnailImageView = [UIImageView new];
        _thumbnailImageView.contentMode = UIViewContentModeScaleAspectFill;
        _thumbnailImageView.clipsToBounds = YES;
    }
    
    return _thumbnailImageView;
    
}

- (UIImageView *)checkView{
    if (!_checkView) {
        _checkView = [[UIImageView alloc] initWithImage:[DKImageResource blueTickImage]];
        _checkView.contentMode = UIViewContentModeCenter;
    }
    return _checkView;
}

- (instancetype)initWithFrame:(CGRect)frame{
    if (self = [super initWithFrame:frame]) {
        self.thumbnailImageView.frame = self.bounds;
        self.thumbnailImageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        
        [self.contentView addSubview:self.thumbnailImageView];
        
        self.checkView.frame = self.bounds;
        self.checkView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        
        [self.contentView addSubview:self.checkView];
        
    }
    return self;
}

- (void)setThumbnailImage:(UIImage *)thumbnailImage{
    super.thumbnailImage = thumbnailImage;
    self.thumbnailImageView.image = thumbnailImage;
}
- (void)setSelected:(BOOL)selected{
    super.selected = selected;
    if (super.selected) {
        self.thumbnailImageView.alpha = 0.5;
        self.checkView.hidden = NO;
    }else{
        self.thumbnailImageView.alpha = 1;
        self.checkView.hidden = YES;
    }
}
/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: Example/DKImagePickerControllerDemo/CustomUIDelegate/CustomGroupDetailImageCell.swift
//
//  CustomGroupDetailImageCell.swift
//  DKImagePickerController
//
//  Created by ZhangAo on 03/01/2017.
//  Copyright © 2017 ZhangAo. All rights reserved.
//

import UIKit
import DKImagePickerController

class CustomGroupDetailImageCell: DKAssetGroupDetailBaseCell {
    
    class override func cellReuseIdentifier() -> String {
        return "CustomGroupDetailImageCell"
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        self.thumbnailImageView.frame = self.bounds
        self.thumbnailImageView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        self.contentView.addSubview(self.thumbnailImageView)
        
        self.checkView.frame = self.bounds
        self.checkView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        self.contentView.addSubview(self.checkView)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override var thumbnailImage: UIImage? {
        didSet {
            self.thumbnailImageView.image = self.thumbnailImage
        }
    }

    internal lazy var _thumbnailImageView: UIImageView = {
        let thumbnailImageView = UIImageView()
        thumbnailImageView.contentMode = .scaleAspectFill
        thumbnailImageView.clipsToBounds = true
        
        return thumbnailImageView
    }()
    
    override var thumbnailImageView: UIImageView {
        get {
            return _thumbnailImageView
        }
    }
    
    fileprivate lazy var checkView: UIImageView = {
        let checkView = UIImageView(image: DKImagePickerControllerResource.blueTickImage())
        checkView.contentMode = .center
        
        return checkView
    }()

    override var isSelected: Bool {
        didSet {
            if super.isSelected {
                self.thumbnailImageView.alpha = 0.5
                self.checkView.isHidden = false
            } else {
                self.thumbnailImageView.alpha = 1
                self.checkView.isHidden = true
            }
        }
    }

}
